content/_code-samples/trade-in-the-decentralized-exchange/js/trade-in-the-dex.js

```diff
@@ -0,0 +1,227 @@
+    // above ours.
+
+    // Unlike above, this time we check for Offers going the same direction as
+    // ours, so TakerGets and TakerPays are reversed from the previous
+    // book_offers request.
+    const orderbook2_resp = await client.request({
+      "command": "book_offers",
+      "taker": wallet.address,
+      "ledger_index": "current",
+      "taker_gets": we_spend,
+      "taker_pays": we_want
+    })
+    console.log(JSON.stringify(orderbook2_resp.result, null, 2))
+
+    // Since TakerGets/TakerPays are reversed, the quality is the inverse.
+    // You could also calculate this as 1/proposed_quality.
+    const offered_quality = BigNumber(we_want.value) / BigNumber(we_spend.value)
+
+    const offers2 = orderbook2_resp.result.offers
+    let running_total2 = 0
+    if (!offers2) {
+      console.log(`No similar Offers in the book. Ours would be the first.`)
+    } else {
+      for (const o of offers2) {
+        if (o.quality <= offered_quality) {
+          console.log(`Existing offer found, funded with ${o.owner_funds}`)
+          running_total2 = running_total2.plus(BigNumber(o.owner_funds))
+        } else {
+          console.log(`Remaining orders are below where ours would be placed.`)
+          break
+        }
+      }
+      console.log(`Our Offer would be placed below at least ${running_total2} ${we_spend.currency}`)
+      if (running_total > 0 && running_total < want_amt) {
+        console.log(`Remaining ${want_amt - running_total} will probably be
+                     placed on top of the order book.`)
+      }
+    }
+  }
+
+  // Send OfferCreate transaction ----------------------------------------------
+
+  // For this tutorial, we already know that TST is pegged to
+  // XRP at a rate of approximately 10:1 plus spread, so we use
+  // hard-coded TakerGets and TakerPays amounts.
+
+  const offer_1 = {
+    "TransactionType": "OfferCreate",
+    "Account": wallet.address,
+    "TakerPays": we_want,
+    "TakerGets": we_spend.value // since it's XRP
+  }
+
+  const prepared = await client.autofill(offer_1)
+  console.log("Prepared transaction:", JSON.stringify(prepared, null, 2))
+  const signed = wallet.sign(prepared)
+  console.log("Sending OfferCreate transaction...")
+  const result = await client.submitAndWait(signed.tx_blob)
+  if (result.result.meta.TransactionResult == "tesSUCCESS") {
+    console.log(`Transaction succeeded: https://testnet.xrpl.org/transactions/${signed.hash}`)
+  } else {
+    throw `Error sending transaction: ${result}`
+  }
+
+  // Check metadata ------------------------------------------------------------
+  // In JavaScript, you can use getBalanceChanges() to help summarize all the
+  // balance changes caused by a transaction.
+  const balance_changes = xrpl.getBalanceChanges(result.result.meta)
+  console.log("Total balance changes:", JSON.stringify(balance_changes, null, 2))
+
+  // Helper to convert an XRPL amount to a string for display
```

Comment:
should this be XRP instead of XRPL?

Reply:
No, it means any currency "Amount" that you may get from an XRP Ledger API.